### DISCUSSION TITLE
How to get the value of the non-grid node

### DISCUSSION THREAD

**Discussion by nanguaxiaofendui** [libMesh/libmesh#3746] [Q&A]
For a problem, after solving the solution vector, how to get the values of other non-grid nodes? Thank you.

**Comment by jwpeterson** [comment]
You can use `System::point_value()` for this purpose if it's just a handful of points, but if you are doing this for "many" points that may not be the most efficient way of doing it. I'm also not sure what is meant by "non-grid nodes". In the libmesh parlance, "Nodes" are explicitly members of the "grid", so it might also help if you explain more what you are trying to do.

**Reply by nanguaxiaofendui** [reply]
Thank you very much. I want to use the average of the solutions obtained from the previous internal iterations as a new right-hand term for iteration, but the resulting "f" is a vector, and I need a function to obtain the corresponding value at the numerical integration point.
Part of the code is as follows:
    MeshTools::Generation::build_cube(mesh, 2, 2, 2, 0, 1., 0, 1., 0, 1., HEX8);
    EquationSystems equation_systems(mesh);
    equation_systems.add_system<LinearImplicitSystem>("TEST");
    equation_systems.get_system("TEST").add_variable("phi", FIRST, MONOMIAL);

for(auto K : make_range(outer_iter)){
    for(auto N : make_range(inner_iter))
    {    
    equation_systems.parameters.set<Real>("Omega_1") = Omega[N][0];
    equation_systems.parameters.set<Real>("Omega_2") = Omega[N][1];
    equation_systems.parameters.set<Real>("Omega_3") = Omega[N][2];
    equation_systems.get_system("TEST").attach_assemble_function(assemble);
    equation_systems.init();
    if(N==0){
    u_bar.assign(equation_systems.n_dofs(),0);
    }
    if(K==0){
    f.assign(equation_systems.n_dofs(),0);
    }
// I need a function f
    equation_systems.parameters.set<std::vector<Real>>("f") = f;
    equation_systems.get_system("TEST").solve();
    NumericVector<Number> &  solution = *equation_systems.get_system("TEST").solution;
    std::vector<Real> v(solution.size());
    solution.localize(v);
    for (auto i : make_range(u_bar.size()))
    u_bar[i] +=Weight[N]*v[i];
  }
    for (auto i : make_range(u_bar.size()))
    f[i] =u_bar[i];
}

//I want "f" is a function
for (unsigned int qp = 0; qp < qface.n_points(); qp++)
      for (std::size_t i = 0; i < dof_indices.size(); i++) {
     Fe(i) += JxW[qp] * f(xyz[qp](0), xyz[qp](1), xyz[qp](2)) * phi[i][qp];
}
